src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/OpenLdapUnsafeMethods.cs

```diff
@@ -0,0 +1,192 @@
+
+        [DllImport(_openldapdll, EntryPoint = "ldap_rename", CharSet = CharSet.Ansi)]
+        public static extern int ldap_rename([In] ConnectionHandle ldapHandle, string dn, string newRdn, string newParentDn, int deleteOldRdn, IntPtr servercontrol, IntPtr clientcontrol, ref int messageNumber);
+
+        [DllImport(_openldapdll, EntryPoint = "ldap_compare_ext", CharSet = CharSet.Ansi)]
+        public static extern int ldap_compare([In] ConnectionHandle ldapHandle, string dn, string attributeName, berval binaryValue, IntPtr servercontrol, IntPtr clientcontrol, ref int messageNumber);
+
+        // Basic Encoding Rules (BER) native functions
+        [DllImport(_openldapdll, EntryPoint = "ber_alloc_t", CharSet = CharSet.Ansi)]
+        public static extern IntPtr ber_alloc(int option);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_init", CharSet = CharSet.Ansi)]
+        public static extern IntPtr ber_init(berval value);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_free", CharSet = CharSet.Ansi)]
+        public static extern IntPtr ber_free([In] IntPtr berelement, int option);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_printf", CharSet = CharSet.Ansi)]
+        public static extern int ber_printf_emptyarg(BerSafeHandle berElement, string format);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_printf", CharSet = CharSet.Ansi)]
+        public static extern int ber_printf_int(BerSafeHandle berElement, string format, int value);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_printf", CharSet = CharSet.Ansi)]
+        public static extern int ber_printf_bytearray(BerSafeHandle berElement, string format, HGlobalMemHandle value, int length);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_printf", CharSet = CharSet.Ansi)]
+        public static extern int ber_printf_berarray(BerSafeHandle berElement, string format, IntPtr value);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_flatten", CharSet = CharSet.Ansi)]
+        public static extern int ber_flatten(BerSafeHandle berElement, ref IntPtr value);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_bvfree", CharSet = CharSet.Ansi)]
+        public static extern int ber_bvfree(IntPtr value);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_bvecfree", CharSet = CharSet.Ansi)]
+        public static extern int ber_bvecfree(IntPtr value);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_scanf", CharSet = CharSet.Ansi)]
+        public static extern int ber_scanf(BerSafeHandle berElement, string format);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_scanf", CharSet = CharSet.Ansi)]
+        public static extern int ber_scanf_int(BerSafeHandle berElement, string format, ref int value);
+
+        [DllImport(_openldapdll, EntryPoint = "ber_scanf", CharSet = CharSet.Ansi)]
+        public static extern int ber_scanf_bitstring(BerSafeHandle berElement, string format, ref IntPtr value, ref int length);
```

Comment:
Did you know that the `length` value here is in bits, not bytes?  I recommend naming the last parameter `bitLength` in both interop files.

```suggestion
        public static extern int ber_scanf_bitstring(BerSafeHandle berElement, string format, ref IntPtr value, ref int bitLength);
```